src/azure-cli/azure/cli/command_modules/acs/decorator.py

```diff
@@ -5811,8 +5990,9 @@ def update_windows_profile(self, mc: ManagedCluster) -> ManagedCluster:
         enable_ahub = self.context.get_enable_ahub()
         disable_ahub = self.context.get_disable_ahub()
         windows_admin_password = self.context.get_windows_admin_password()
+        enable_windows_gmsa = self.context.get_enable_windows_gmsa()
 
-        if any([enable_ahub, disable_ahub, windows_admin_password]) and not mc.windows_profile:
+        if any([enable_ahub, disable_ahub, windows_admin_password, enable_windows_gmsa]) and not mc.windows_profile:
```

Comment:
Users can only change the configuration of gmsa when they update the cluster if they set the windows profile when the cluster is created.

If the user does not set the windows-related configuration when creating the cluster, the configuration of gmsa cannot be changed through update.